# **Pipelines @ Comparative genomics of <i>Xylella fastidiosa</i> explores determinants of host-specificity and expands its mobile genetic elements repertoire and defense systems.**

At these following pages we present 5 pipelines of analysis that are present at the paper of MDPI microorganisms' paper of Uceda et al. 2022.

---

## RELEASE

Version 1.0.0 - Mar 17, 2022.

Available from <https://github.com/gucedac/uceda-campos-etal-2022-microorganisms>. 

---

## 1. DATA COLLECTION, CURATION, AND ANNOTATION 

Here we describe main commands used for genome analysis, annotation, and metadata inclusion.

---

## 2. GENOME COMPARISONS, PHYLOGENOMICS, AND FUNCTIONAL ANALYSIS

Here we describe how we made genome comparisons, used the data to make the phylogenetic tree and include a command for a COG analysis.

---

## 3. MAPPING SCORE CALCULATION (SMAP) 

Here we describe an algorithm in-house developed to map genetic traits. In the case of our paper, the algorithm was adopted to map genes of different <i>Xylella fastidiosa</i> strains to their respective host-trees.
We also include some examples to illustrate the algorithm's usage.

---

## 4. MOBILE GENETIC ELEMENTS PREDICTION

Here we describe all the prediction tools we used to detect MGE (prophages, genomic island and insertion sequences) as well as the steps we took to integrate the mobilome.

---

## 5. ANTI-MGE DEFENSE SYSTEMS

Here we describe all the prediction tools we used to detect anti-MGE defense systems.

---

## CITATION

If you find any of the describe pipelines useful in your research, please cite the following reference according to the scripts you use:

- Uceda-Campos, G., Feitosa-Junior, O.R., Santiago, C. R. N., Pierry, P.M., Zaini, P.A., de Santana, W.O., Mar-tins-Jr, J., Barbosa, D., Digiampietri, L. A., Setubal, J.C., da Silva, A.M. Comparative genomics of Xylella fastidiosa explores determinants of host-specificity and expands its mobile genetic elements repertoire Microorganisms 2022, 10, x. https://doi.org/10.3390/xxxxx (under peer review).

---

## ISSUES AND REQUESTS

If you experience any issues or would like to see support for an additional feature implemented in the pipelines described at this page, please file a request in the GitHub issue tracker or email it to the developer. Feel free to contact the developer with any further questions regarding these scripts. You can reach him at the e-mail: gucedac@ime.usp.br.

---

## LICENSES

Copyright Â© 2022 Guillermo Uceda 

These pipelines are free softwares: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, or (at your option) any later version.

These pipelines are distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License along with this program.  If not, see <https://www.gnu.org/licenses/>.

The dataset included with these pipelines are licensed under the Creative Commons Attribution-ShareAlike 4.0 International License. To view a copy of this license, visit http://creativecommons.org/licenses/by-sa/4.0/.
